### PR TITLE
Rename --max option to --max-tests-per-plan with deprecation

### PR DESCRIPTION
**Pull request by happz**
The --max option is now deprecated in favor of --max-tests-per-plan for better clarity. The old option still works but shows a deprecation warning.

🤖 Polished by [Claude Code](https://claude.ai/code)

Pull Request Checklist

* [x] implement the feature

---

Reviewed-by: [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: tmt/plugins/plan_shapers/max_tests.py [hunk 1]
```diff
@@ -21,24 +21,36 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
...
+            ),
             option(
                 '--max',
                 metavar='N',
-                help='Split plans to include N tests at max.',
+                help='Split every plan to include N tests at maximum.',
                 type=int,
                 default=-1,
-            )
+                deprecated=Deprecated('1.57', hint='use ``--max-tests-per-plan`` instead'),
```
Comment (LecrisUT) [inline tmt/plugins/plan_shapers/max_tests.py]: Postponing this until 1.57? Seems like a trivial change to merge at anytime.

Comment (happz) [reply tmt/plugins/plan_shapers/max_tests.py]: I agree, anytime - once it gets reviewed, and that's often unpredictable. Plenty of such "anytime" PRs in https://github.com/orgs/teemtee/projects/1/views/6.

I can change the version once it becomes clear which milestone would this get into.

**Review by vaibhavdaren — APPROVED** [review]

**Comment by happz** [comment]
As mentioned in https://github.com/teemtee/tmt/pull/3234#issuecomment-3187117514, the `/tests/execute/framework/beakerlib` failure is not related to our changes.

File: tmt/plugins/plan_shapers/max_tests.py (hunk 1 above)
Comment (happz) [reply tmt/plugins/plan_shapers/max_tests.py]:
```suggestion
                deprecated=Deprecated('1.55', hint='use ``--max-tests-per-plan`` instead'),
```
